client/src/context.js

```diff
@@ -0,0 +1,125 @@
+import React, { createContext, useContext, useReducer } from "react";
+import { authApiUrl, ingestApiUrl, loginUrl } from "./routes";
+import PropTypes from "prop-types";
+
+const fetchContextParams = {
+    token: sessionStorage.getItem("token") ?? null,
+};
+const FetchContext = createContext({});
+
+const fetchReducer = (state, { type, payload }) => {
+    switch (type) {
+        case "UPDATE_TOKEN":
+            sessionStorage.setItem("token", payload);
+            return {
+                ...state,
+                token: payload,
+            };
+        default:
+            console.error("action type does not exist");
```

Comment:
Would it be helpful to actually output the value of type as part of the error message in this case, so if it ever happens we know which action was requested?

Reply:
Made change